src/main/java/ama/post/PostController.java

```diff
@@ -118,14 +118,13 @@ public String replyToComment(HttpServletRequest request, @PathVariable(value="id
         User user = userRepository.findByUsername(username);
         if (user != null && context != null && reply != null && reply != "") {
             CommentPost replyPost = new CommentPost(user, context, reply);
-
             commentPostRepository.save(replyPost);
+            return "redirect:" + request.getHeader("Referer");
         }
-        String referer = request.getHeader("Referer");
-        return "redirect:" + referer;
+        return "pageNotFound";
```

Comment:
are you sure we want to do this? This means if someone hits reply with an empty text box they will be shown a 404 page. Seems like the best experience in this case would be to always redirect to the original page.